internal/app/users/service.go

```diff
@@ -303,11 +303,24 @@ func (us *service) ListUsers(ctx context.Context, reqData dto.ListUsersReq) (res
 		return
 	}
 
+	userId := ctx.Value(constants.UserId)
+
 	var users []dto.UserDetails
 
 	for _, dbUser := range dbResp {
-		user := mapDbUserToUserListResp(dbUser)
-		users = append(users, user)
+		if reqData.Self {
+			if dbUser.Id != userId {
+				user := mapDbUserToUserListResp(dbUser)
+				users = append(users, user)
+			}
+		} else {
+			user := mapDbUserToUserListResp(dbUser)
+			users = append(users, user)
+		}
+	}
+
+	if reqData.Self && totalCount > 0 {
+		totalCount = totalCount - 1
```

Comment:
What is this logic for?

Reply:
If the logged in person is excluded from the user list, the total count must be reduced by one

Reply:
sorry my bad, actually I didn't select all the lines for which I was asking. check the suggestion